test/index.js

```diff
@@ -1892,8 +1892,62 @@ describe('Nightmare', function () {
       nightmare = Nightmare({ electronPath: require('electron') });
       nightmare.should.be.ok;
     });
+
+    it('should allow to use external Promise', function*() {
+      nightmare = Nightmare({ Promise: require('bluebird') });
```

Comment:
maybe the require('bluebird') return can be assigned to a variable at the beginning of this generator function. Although I know that require is always cached, I wonder if it would look nicer if it's in a variable.